data/auxiliary/gather/ldap_query/ldap_queries_default.yaml

```diff
@@ -10,19 +10,42 @@ queries:
       - cACertificateDN
       - dNSHostname
       - certificateTemplates
-  - action: ENUM_ADCS_CERT_TEMPLATES
-    description: 'Enumerate ADCS certificate templates.'
+      - objectGUID
+      - caCertificate
+    referemces:
+      - https://aaroneg.com/post/2018-05-15-enterprise-ca/
+  - action: ENUM_ADCS_ESC1_VULN_TEMPLATES
+    description: 'Enumerate ADCS ESC1 vulnerable certificate templates.'
     base_dn_prefix: 'CN=Certificate Templates,CN=Public Key Services,CN=Services,CN=Configuration'
-    filter: '(objectClass=pkicertificatetemplate)'
+    filter: '(&(objectclass=pkicertificatetemplate)(!(mspki-enrollment-flag:1.2.840.113556.1.4.804:=2))(|(mspki-ra-signature=0)(!(mspki-ra-signature=*)))(|(pkiextendedkeyusage=1.3.6.1.4.1.311.20.2.2)(pkiextendedkeyusage=1.3.6.1.5.5.7.3.2) (pkiextendedkeyusage=1.3.6.1.5.2.3.4))(mspki-certificate-name-flag:1.2.840.113556.1.4.804:=1))'
     attributes:
       - cn
       - name
       - displayName
+      - msPKI-Cert-Template-OID
```

Comment:
Should add in the `certificateTemplates` and `caCertificate` fields here.

Reply:
Actually even more important is the security descriptor attribute which will let us know who can actually use the certificate for enrollment and who can edit the certificate template.

Reply:
Ignoring the security descriptor attribute for now since that will likely require additional parsing. However its still a valid point here.